hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreScanner.java

```diff
@@ -934,6 +950,39 @@ public void updateReaders(List<HStoreFile> sfs, List<KeyValueScanner> memStoreSc
     // Let the next() call handle re-creating and seeking
   }
 
+  // Implementation of ChangedReadersObserver
+  // reset current scanners
+  @Override
+  public void updateReaders(List<HStoreFile> storeFiles) throws IOException {
+    if (CollectionUtils.isEmpty(storeFiles)) {
+      return;
+    }
+    try {
+      if (this.closing) {
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("StoreScanner already closing. There is no need to re-create scanners");
+        }
+        return;
+      }
+      SCANNER_RESET_LOCK.lock();
+      // let next(), seek() take care of resetting scanners
+      this.scannersToBeReset = true;
+      final boolean usePRead = get || scanUsePread;
+      final boolean isCompaction = false;
+      final boolean includeMemstoreScanner = false;
+      List<KeyValueScanner> scanners = store.getScanners(storeFiles, cacheBlocks, get, usePRead,
```

Comment:
Shall we unify the code in both updateReaders? also the reopenCode- if not it will lead to duplicate code and logic is almost similar?

Reply:
Sure, we can take individual locks in each method and then retrieval of list of KeyValueScanner can be a common utility method